### DISCUSSION TITLE
trunk-scanner System Tagging

### DISCUSSION THREAD

**Discussion by HeroesLament** [chuot/rdio-scanner#353] [Q&A]
Hello Rdio folks! So, when using Rdio with trunk-scanner, I get this in the logs:

2023/05/22 14:03:17 dirwatch.walkdir: stat .: no such file or directory
2023/05/22 14:03:17 dirwatch.walkdir: stat .: no such file or directory
2023/05/22 14:03:17 dirwatch.ingest: no system, /var/lib/trunk-recorder/ALMR/2023/5/22/31205-1684788807_155850000-call_7.json
2023/05/22 14:03:17 dirwatch.ingest: no system, /var/lib/trunk-recorder/ALMR/2023/5/22/31205-1684788821_155850000-call_8.json

And looking at the trunk-scanner metadata:

{
"freq": 155850000,
"start_time": 1684788821,
"stop_time": 1684788824,
"emergency": 0,
"encrypted": 0,
"call_length": 4,
"talkgroup": 31205,
"talkgroup_tag": "",
"talkgroup_description": "",
"talkgroup_group_tag": "",
"talkgroup_group": "",
"audio_type": "digital",
"short_name": "ALMR",
"freqList": [ {"freq": 155850000, "time": 1684788821, "pos": 0.00, "len": 3.60, "error_count": "34", "spike_count": "7"} ],
"srcList": [ {"src": 3123564, "time": 1684788821, "pos": 0.00, "emergency": 0, "signal_system": "", "tag": ""} ]
}

There's really nothing along the lines of system here. How should we marry a trunk-scanner recording to a system in rdio?

**Comment by rdmcvicar** [comment]
Just checking back on this in case it's not been answered for you yet, HL.  Is this metadata from a .json created by trunk-recorder (versus -scanner)?  If -recorder, I can offer a couple steps to take.

**Reply by HeroesLament** [reply]
Yes, this was from recorder, I see I actually incorrectly titled my post.

**Comment by rdmcvicar** [comment]
While trunk-recorder can write audio and metadata files, and while the original post refers to a rdio-scanner dirwatch, I personally use trunk-recorder's rdio-scanner plugin.

With this method, the marrying is done by:

1.  On the rdio-scanner side:

Make sure you have a system set up.  It looks like in this case, we're monitoring Alaska Land Mobile Radio (at one of the four sites that use 155.850 MHz as a voice channel...and given the talkgroup number I'll guess [High Mtn - Ketchikan](https://www.radioreference.com/db/site/5737)).  [ALMR](https://www.radioreference.com/db/sid/2447)'s system ID is hex142, or decimal322.  So 322 would be a handy "Id" to use on the rdio-scanner side (Config > Systems).  I realize some folks want each site of a multi-site system to be it's own rdio-scanner system, but I prefer running all sites into a single system and employing various means to limit duplicates.  If your rdio-scanner server handles other ALMR sites and you want them to be separate, you may wish to use a different system numbering scheme.

Also set up an API key (Config > ApiKey).  I choose to make sure the system name matches the above, but don't believe that's required.  

We'll pretend you're using system name "ALMR" and you've generated API key: "12345678-1234-1234-1234-abcdefghijkl".  And we'll pretend your rdio-scanner server lives at http://1.2.3.4.

2.  On the trunk-recorder side:

Including a [rdio-scanner plugin object](https://github.com/robotastic/trunk-recorder/blob/master/docs/CONFIGURE.md#rdio-scanner-plugin) that looks like this:

 {
          "name": "rdioscanner_uploader",
          "library": "librdioscanner_uploader.so",
          "server": "http://1.2.3.4",
          "systems": [{
                  "shortName": "ALMR",
                  "apiKey": "12345678-1234-1234-1234-abcdefghijkl",
                  "systemId": 322
          }

Hopefully this helps, and/or others quickly correct anything I'm doing wrong here.  But this method works well for me.

If you or anyone else need a rdio-scanner server to send trunk-recorder traffic to and/or vice-versa, please feel free to reach out.  I enjoy trading sites / systems.

Rich McVicar
rdmcvicar tds.net
Middleton, WI
